tests/h/settings_test.py

```diff
@@ -98,3 +98,63 @@ def test_mandrill_settings():
 
 def test_mandrill_settings_unset():
     assert settings.mandrill_settings({}) is None
+
+
+class TestSettingsManager(object):
+    def test_does_not_emit_warnings_when_deprecated_setting_is_not_used(self, log):
+        settings_manager = settings.SettingsManager({}, {})
+        settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+
+        assert not log.warn.called
```

Comment:
I think you might want to use `log.warn.assert_not_called()` instead here because you'll get a better failute message if it was called. See <https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called>. Maybe test this though: it's possible that pytest intercepts the basic `assert not log.warn.called` and replaces it with a better message.

Reply:
Oh, but wait: actually it shouldn't be necessary to mock `log` at all because it's better to use `pytest.warns`. See: https://docs.pytest.org/en/latest/warnings.html#assertwarnings

Reply:
While I would love to use pytest warn it's proving not so easy. Here's a link to what I believe is the problem I'm running into. https://github.com/pytest-dev/pytest/issues/2917 Maybe you can figure it out but I'm not sure it's worth spending several hours for such little reward. It seems like the default config for the filters in py2.7 ignores warnings and so we have to override the filters but I tried that and it still didn't work so I'm at a loss. Not sure it's worth blocking the issue for.